pledgeindia/pledge/models.py

```diff
@@ -0,0 +1,101 @@
+from django.contrib.auth.base_user import AbstractBaseUser, BaseUserManager
+from django.contrib.auth.models import PermissionsMixin
+from django.db import models
+from django.utils.translation import ugettext_lazy as _
+
+# Create your models here.
+
+class UserManager(BaseUserManager):
+    use_in_migrations = True
+
+    def _create_user(self, email, password, **extra_fields):
+        """
+        Creates and save a user with given email and password.
+        """
+
+        if not email:
+            raise ValueError('Email is Required')
+        
+        email = self.normalize_email(email)
+        user = self.model(email=email, **extra_fields)
+        user.set_password(password)
+        user.save(using=self._db)
+        return user
+
+    def create_user(self, email, password=None, **extra_fields):
+        extra_fields.setdefault('is_superuser', False)
+        return self._create_user(email, password, **extra_fields)
+
+    def create_superuser(self, email, password, **extra_fields):
+        extra_fields.setdefault('is_superuser', True)
+        
+        if extra_fields.get('is_superuser') is not True:
+            raise ValueError('Superuser must have is_superuser=True')
+        
+        return self._create_user(email, password, **extra_fields)
+
+
+class User(AbstractBaseUser, PermissionsMixin):
+    """User Model"""
+    gender_choices = (
+        ('male', 'Male'),
+        ('female', 'Female'),
+        ('other','Other'),
+    )
+
+    email = models.EmailField(_('email address'), unique=True)
+    first_name = models.CharField(_('first name'), max_length=30, blank=True)
+    last_name = models.CharField(_('last name'),  max_length=30, blank=True)
+    date_joined = models.DateField(_('date joined'), auto_now=True)
+    is_active = models.BooleanField(_('active'), default=True)
+
+    gender = models.TextField(_('gender'), choices=gender_choices, null=True, blank=True)
+
+    objects = UserManager()
+
+    USERNAME_FIELD = 'email'
+    REQUIRED_FIELDS = ['first_name']
+
+    class Meta:
+        verbose_name = _('user')
+        verbose_name_plural = _('users')
+
+    def __str__(self):
+        return self.get_full_name()
+
+    def get_full_name(self):
+        '''
+        Returns the first_name plus the last_name, with a space in between.
+        '''
+        full_name = '%s %s' % (self.first_name, self.last_name)
+        return full_name.strip()
+
+    def get_short_name(self):
+        '''
+        Returns the short name of user.
+        '''
+        return self.first_name
+
+
+class Pledge(models.Model):
+    """Pledge Model"""
+
+    title = models.CharField(_('title'), max_length = 100)
+    description = models.TextField(_('description'),null=True)
+
+    class Meta:
+        verbose_name = _('pledge')
+        verbose_name_plural = _('pledges')
+
+    def __str__(self):
+        return self.title
+
+class User_Pledges(models.Model):
```

Comment:
have a base timestamped uuid model for all our models ?
https://github.com/Fueled/django-init/blob/master/%7B%7Bcookiecutter.github_repository%7D%7D/%7B%7Bcookiecutter.main_module%7D%7D/base/models.py#L20